### ISSUE TITLE
Rust can't find its libraries

### ISSUE THREAD

**Issue by vberger** [Heather/gentoo-rust#41]
I recompiled rust since last update of the ebuild, and if I try to run even `rustc --help`, it fails with:
```
rustc: error while loading shared libraries: librustc_driver-gentoo-git.so: cannot open shared object file: No such file or directory
```
If I supply it the path like this:
```
LD_LIBRARY_PATH="/usr/lib/rust-9999/rust-9999/rustlib/x86_64-unknown-linux-gnu/lib/" rustc
```
It will work, but trying to compile anything fails with:
```
src/lib.rs:1:1: 1:1 error: can't find crate for `std`
```

Compiled with useflags `vim-syntax -clang -debug -emacs -libcxx -system-llvm -zsh-completion`.

**Comment by suhr** [comment]
> Is this bug should be sent upstream?

Yeah, why not?

**Comment by williamh** [comment]
@Heather:
Can  you please re-open this?

The "fix" we are using is a kludge that moves the files around, so we do not install them the way upstream does. In the longrun this kind of distro-specific kludge is not a good idea, so I would like to re-visit this issue.

hanks much,

William

**Comment by jauhien** [comment]
Yes, we should have more look at this problem. Reopening.

**Comment by jauhien** [comment]
/offtop
@Heather: could you, please, add @williamh as collaborator to this repo as he is a co-maintainer of Rust in the main tree?

**Comment by Heather** [comment]
@jauhien sure

**Comment by williamh** [comment]
Ok folks, I think I know how to rework this and clean up slots at the same time. However, I have a question about slots.

Do we need multiple *versions* of rust installed, or is multiple *release channels* enough?
Remember their release channels are going to be "beta", "stable", "nightly" and "dev".
I think we could just go with release channels, because things that will run with an old release channel, e.g. stable, will run with newer code.

What does everyone else think?

**Comment by jauhien** [comment]
We need release channels slots, I think. Version slots were needed only because of breaking changes in pre-alpha rust.

**Comment by williamh** [comment]
@Jauhien:
I think this issue is related to us trying to support slots, which upstream doesn't support yet.
There is an open issue upstream for allowing multiple versions of rust to be installed, but it has not been resolved yet. Honestly, I think our support for slots is going to be fragile at best until upstream resolves rust-lang/rust#19263.

**Comment by jauhien** [comment]
See https://github.com/gentoo/gentoo-rust/issues/5
